### PR TITLE
HL-686

### PR DESCRIPTION
**Pull request by bitcoinbrisbane**
[Link to ticket](url)

- Fixes to the oracle to stop DOS
- More tests
- Clean up settle logic

### CODE REVIEW

File: contracts/Market.sol [hunk 1]
```diff
@@ -61,9 +61,10 @@ contract Market is IMarket, Ownable, ERC721 {
 	mapping(address => bool) private _signers;
 
 	// Race result constants
-    uint8 internal constant WINNER = 0x01;
-    uint8 internal constant LOSER = 0x02;
-    uint8 internal constant SCRATCHED = 0x03;
+	uint8 public constant NULL = 0x00;
+    uint8 public constant WINNER = 0x01;
```
Comment (bazmatic) [inline contracts/Market.sol:65]: Indent seems off

File: tests/market.oracle.test.ts [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+			const propositionId1 = makePropositionId(marketId, 1);
+			const propositionId2 = makePropositionId(marketId, 2);
+
+			const nonce = "1";
+			const currentTime = await time.latest();
+			// Assume race closes in 1 hour from now
+			const close = currentTime + 3600;
+			const end = 1000000000000;
+
+			// Bet on proposition 1 (looser)
```
Comment (bazmatic) [inline tests/market.oracle.test.ts:172]:
```suggestion
			// Bet on proposition 1 (loser)
```

File: tests/market.test.ts [hunk 3]
```diff
@@ -854,40 +854,44 @@ describe("Market", () => {
...
-			expect(inPlay).to.equal(0);
-
-			// Everything should be as it was before
-			const balance = await underlying.balanceOf(bob.address);
-			const endVaultBalance = await underlying.balanceOf(vault.address);
-			const endMarketBalance = await underlying.balanceOf(market.address);
-			expect(balance).to.equal(bobBalance);
-			expect(vaultBalance).to.equal(endVaultBalance);
-			expect(marketBalance).to.equal(endMarketBalance);
+			// const newNftBalance = await market.balanceOf(bob.address);
```
Comment (bazmatic) [inline tests/market.test.ts:876]: Did you mean to leave this commented out?

File: tests/market.test.ts [hunk 4]
```diff
@@ -1147,8 +1159,8 @@ describe("Market", () => {
 
 			await market.settleMarket(formatBytes16String(marketId));
 
-			inPlayCount = await market.getInPlayCount();
-			expect(inPlayCount).to.equal(0);
+			// inPlayCount = await market.getInPlayCount();
```
Comment (bazmatic) [inline tests/market.test.ts:1162]: Uncomment?

**Review by bazmatic — CHANGES_REQUESTED** [review]
Couple minor things

File: contracts/Market.sol (hunk 1 above)
Comment (bitcoinbrisbane) [reply contracts/Market.sol:65]: Looks fine to me.  I linted too.

File: tests/market.test.ts (hunk 3 above)
Comment (bitcoinbrisbane) [reply tests/market.test.ts:876]: Nope.  Fixed.

File: tests/market.test.ts (hunk 4 above)
Comment (bitcoinbrisbane) [reply tests/market.test.ts:1162]: Nope.  Fixed.

**Review by bazmatic — APPROVED** [review]

**Pull request merged**